### ISSUE TITLE
Use of strings.HasSuffix in core::0191::proto-package rule is overly permissive

### ISSUE THREAD

**Issue by apasel422** [googleapis/api-linter#868]
For example, the following test case erroneously passes:

```go
{"InvalidSuffix", "hgoogle/library/v1/library.proto", "google.library.v1", testutils.Problems{{Message: "directory structure"}}},
```

In other words, `strings.HasSuffix` alone doesn't require that the first path component matches exactly.
